### PR TITLE
feat(mqtt source, mqtt sink): add MQTT v5 protocol support

### PR DESCRIPTION
**Pull request by vitalvas**
## Summary

- Add MQTT v5 protocol support to the `mqtt` source and `mqtt` sink, selectable via a new `protocol_version` option (defaults to `v311` for backward compatibility).
- Expose v5 publish properties (payload format indicator, message expiry, topic alias, response topic, correlation data, content type, user properties) on the sink under a new `publish_properties` config section, and surface incoming v5 properties as source event metadata.
- Add end-to-end acknowledgements to the `mqtt` source: when enabled, the source switches to MQTT manual ack mode and only sends `PubAck` to the broker once Vector has delivered the event downstream, providing at-least-once semantics for both v3.1.1 and v5.
- Derive the sink's accepted input types from the configured encoder so logs, metrics, and traces are all transportable over MQTT (default `json` encodes all three; use `native_json` for lossless Vector-to-Vector p ipelines). Mis-routed event types are now rejected at config build time instead of silently dropped by the codec.
- Emit `open_connections` and `connection_shutdown_total` telemetry from both source and sink, plus richer `component_errors_total` coverage for connection, subscribe, and acknowledgement failures.
- Consolidate the v3 and v5 connection-building code into a shared `common::mqtt::build_connector` to keep the source and sink in sync.

## Vector configuration

```yaml
sources:
  my_mqtt_source:
    type: mqtt
    host: "mqtt.example.com"
    port: 1883
    topic: "events/#"
    protocol_version: v5
    acknowledgements: true
    connect_properties:
      session_expiry_interval: 300

sinks:
  my_mqtt_sink:
    type: mqtt
    inputs: ["my_mqtt_source"]
    host: "mqtt.example.com"
    port: 1883
    topic: "events/{{ .namespace }}"
    protocol_version: v5                                                                                                                                                                                                  publish_properties:
      content_type: "application/json"
      message_expiry_interval: 3600
      user_properties:
        source: vector
    encoding:
      codec: json
```

## How did you test this PR?

- New unit tests for connector building (v3/v5, manual_acks propagation, credential and client_id validation), publish-property conversion (all fields, boundary values), QoS conversions, source metadata application (v3 and v5, all 7 v5 fields, no-op on non-log events), and direction tagging on connection errors. 25 mqtt unit tests pass.
- Existing v3.1.1 integration tests continue to pass.
- Manual verification that the default `protocol_version` remains `v311` and existing configurations behave unchanged.

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Dependencies
- [ ] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?

- [ ] Yes
- [x] No

## Does this PR include user facing changes?

- [x] Yes. Please add a changelog fragment based on our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [ ] No. A maintainer will apply the `no-changelog` label to this PR.

## Notes

- Please read our [Vector contributor resources](https://github.com/vectordotdev/vector/tree/master/docs#getting-started).
- Do not hesitate to use `@vectordotdev/vector` to reach out to us regarding this PR.
- Some CI checks run only after we manually approve them.
  - We recommend adding a `pre-push` hook, please see [this template](https://github.com/vectordotdev/vector/blob/master/CONTRIBUTING.md#Pre-push).
  - Alternatively, we recommend running the following locally before pushing to the remote branch:
    - `make fmt`
    - `make check-clippy` (if there are failures it's possible some of them can be fixed with `make clippy-fix`)
    - `make test`
- After a review is requested, please avoid force pushes to help us review incrementally.
  - Feel free to push as many commits as you want. They will be squashed into one before merging.
  - For example, you can run `git merge origin master` and `git push`.
- If this PR introduces changes Vector dependencies (modifies `Cargo.lock`), please
  run `make build-licenses` to regenerate the [license inventory](https://github.com/vectordotdev/vrl/blob/main/LICENSE-3rdparty.csv) and commit the changes (if any). More details on the [dd-rust-license-tool](https://crates.io/crates/dd-rust-license-tool).

### CODE REVIEW

**Comment by vitalvas** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by vitalvas** [comment]
@codex review

File: website/cue/reference/components/sinks/generated/mqtt.cue [hunk 1]
```diff
@@ -37,6 +37,44 @@ generated: components: sinks: mqtt: configuration: {
 		required:    false
 		type: string: {}
 	}
+	connect_properties: {
+		description: "MQTT v5 connection properties. Only used when protocol_version is v5."
```
Comment (rtrieu) [inline website/cue/reference/components/sinks/generated/mqtt.cue]:
```suggestion
		description: "MQTT v5 connection properties. Only applicable when `protocol_version` is `v5`."
```

File: website/cue/reference/components/sources/mqtt.cue [hunk 2]
```diff
@@ -61,45 +61,162 @@ components: sources: mqtt: {
...
-	how_it_works: components._amqp.how_it_works
+	how_it_works: {
+		topic_subscription: {
+			title: "Topic subscriptions"
+			body: """
+				The `topic` option accepts either a single topic string or a list of topics.
+				All subscriptions are made at MQTT QoS 1 (AtLeastOnce), and MQTT wildcards
+				are supported:
+
+				- `+` matches exactly one topic level (for example `sensors/+/temperature`)
```
Comment (rtrieu) [inline website/cue/reference/components/sources/mqtt.cue]:
```suggestion
				- `+` matches exactly one topic level (for example ,`sensors/+/temperature`)
```

File: website/cue/reference/components/sources/mqtt.cue [hunk 3]
```diff
@@ -61,45 +61,162 @@ components: sources: mqtt: {
...
+				- `#` matches zero or more trailing levels (for example `sensors/#`)
```
Comment (rtrieu) [inline website/cue/reference/components/sources/mqtt.cue]:
```suggestion
				- `#` matches zero or more trailing levels (for example ,`sensors/#`)
```

File: website/cue/reference/components/sources/mqtt.cue [hunk 4]
```diff
@@ -61,45 +61,162 @@ components: sources: mqtt: {
...
+					required:    false
+					common:      false
+					type: string: {
+						default: null
+						examples: ["abc123"]
+						syntax: "literal"
+					}
+				}
+				payload_format_indicator: {
+					description: "MQTT v5 payload format indicator. 0 indicates unspecified bytes and 1 indicates UTF-8 encoded data. Absent for v3.1.1 messages or v5 messages without the property set."
```
Comment (rtrieu) [inline website/cue/reference/components/sources/mqtt.cue]:
```suggestion
					description: "MQTT v5 payload format indicator. 0 indicates unspecified bytes, and 1 indicates UTF-8 encoded data. Absent for v3.1.1 messages or v5 messages without the property set."
```

File: website/cue/reference/components/sinks/mqtt.cue [hunk 5]
```diff
@@ -63,9 +64,59 @@ components: sinks: mqtt: {
...
+			summary:      true
+		}
+		traces: true
+	}
+
+	how_it_works: {
+		protocol_versions: {
+			title: "MQTT 3.1.1 and 5.0 support"
+			body: """
+				The sink supports both MQTT 3.1.1 (default) and MQTT 5.0, selected via the
```
Comment (rtrieu) [inline website/cue/reference/components/sinks/mqtt.cue]:
```suggestion
				The sink supports both MQTT 3.1.1 (default) and MQTT 5.0, selected using the
```

File: website/cue/reference/components/sinks/mqtt.cue [hunk 6]
```diff
@@ -63,9 +64,59 @@ components: sinks: mqtt: {
...
+				`protocol_version` configuration option. MQTT 5.0 unlocks additional features
+				such as message expiry, topic aliases, response topics, correlation data,
+				content type, payload format indicators, and user properties.
+				"""
+		}
+		publish_properties: {
+			title: "MQTT v5 publish properties"
+			body: """
+				When `protocol_version` is set to `v5`, the sink can attach MQTT 5.0 publish
+				properties to outgoing messages via the `publish_properties` section. Supported
```
Comment (rtrieu) [inline website/cue/reference/components/sinks/mqtt.cue]:
```suggestion
				properties to outgoing messages through the `publish_properties` section. Supported
```

**Review by rtrieu — APPROVED** [review]
some minor suggestions to comply with our style guide

**Comment by vitalvas** [comment]
@rtrieu Thanks for the suggestions. Applied them in eee8b3e7793cdfc0590e12884b1a60be3d2d0ea5
